services/paymaster/src/api.rs

```diff
@@ -405,24 +429,76 @@ struct SponsorPubkeyQuery {
     #[serde(default)]
     /// Domain to request the sponsor pubkey for
     domain: Option<String>,
+    #[serde(default)]
+    #[param(value_type = String)]
+    /// Index of the sponsor to select. Can be a number or the string "autoassign" which may give you a different sponsor every time
+    index: Option<IndexSelector>,
+}
+
+#[derive(serde::Deserialize)]
+#[serde(try_from = "String")]
+enum IndexSelector {
+    Autoassign,
+    Index(u8),
+}
+
+impl TryFrom<String> for IndexSelector {
+    type Error = String;
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        if value == "autoassign" {
+            Ok(IndexSelector::Autoassign)
+        } else {
+            let i: u8 = value.parse().map_err(|_| {
+                format!(
+                    "Invalid index value: {value}. Use a number between 0 and 255 or 'autoassign'"
+                )
+            })?;
+            Ok(IndexSelector::Index(i))
+        }
+    }
 }
 
 #[utoipa::path(get, path = "/sponsor_pubkey", params(SponsorPubkeyQuery))]
 async fn sponsor_pubkey_handler(
     State(state): State<Arc<ServerState>>,
     origin: Option<TypedHeader<Origin>>,
-    Query(SponsorPubkeyQuery { domain }): Query<SponsorPubkeyQuery>,
+    Query(SponsorPubkeyQuery { domain, index }): Query<SponsorPubkeyQuery>,
 ) -> Result<String, ErrorResponse> {
     let domain = get_domain_name(domain, origin)?;
     let domains_guard = state.domains.load();
     let domain_state = get_domain_state(&domains_guard, &domain)?;
     let DomainState {
         domain_registry_key: _,
-        sponsor,
+        sponsors,
         enable_preflight_simulation: _,
         tx_variations: _,
+        next_autoassigned_sponsor_index,
     } = domain_state;
-    Ok(sponsor.pubkey().to_string())
+
+    let sponsor_index = if let Some(selector) = index {
+        match selector {
+            IndexSelector::Autoassign => {
+                next_autoassigned_sponsor_index.fetch_add(1, Ordering::Relaxed) % sponsors.len()
```

Comment:
i don't think this is a good feature. i would rather default to the basic pubkey (seed 0) and show different pubkeys if requested by the app. i don't get why a user would ever want this non-deterministic behavior from this GET endpoint